components/server/src/code-sync/code-sync-service.ts

```diff
@@ -343,17 +269,7 @@ export class CodeSyncService {
             }
         }
 
-        let resourceRev: string | undefined = ref;
-        if (resourceRev !== fromTheiaRev) {
-            resourceRev = (await this.db.getResource(req.user.id, resourceKey, resourceRev, collection))?.rev;
-        }
-        if (
-            !resourceRev &&
-            !collection &&
-            (resourceKey === SyncResource.Extensions || resourceKey === SyncResource.Settings)
-        ) {
-            resourceRev = fromTheiaRev;
-        }
+        const resourceRev = (await this.db.getResource(req.user.id, resourceKey, ref, collection))?.rev;
         if (!resourceRev) {
```

Comment:
`!resourceRev || resourceRev === fromTheiaRev` ? 

some clients can already cache it?

Reply:
although it should fail like that anyway after trying to access content storage

Reply:
Would be awesome, if you'd double-check this area. I don't want to break users.

Reply:
@jeanp413 could you double check please, on first glance it looks alright

Reply:
yep looks good